### ISSUE TITLE
ImageService v2: Error Creating Properties

### ISSUE THREAD

**Issue by jtopjian** [gophercloud/gophercloud#259]
The combination of current [JSON tags](https://github.com/gophercloud/gophercloud/blob/c5c590a7ee87267c978586fdd1655db0dd4b6e05/openstack/imageservice/v2/images/requests.go#L102) is causing the JSON body to not be rendered properly to the API.

The following isn't JSON, but a Python dict I dumped from the request on the server:

```python
{
  'name': 'Rancher TerraformAccTest', 
  'container_format': 'bare', 
  'visibility': 'private', 
  'protected': False, 
  'disk_format': 'qcow2',
    '-': {'architecture': 'x86_64'}, 
  'architecture': 'x86_64'
}
```

Note the last two entries.

I think `omitempty` can be removed since the `ToImageCreateMap` function checks for nil anyway? If so, I have a PR ready to go for that along with the start of acceptance tests.

**Comment by jrperritt** [comment]
Confirmed. Please go ahead.

**Comment by jrperritt** [comment]
closed by #264
